### ISSUE TITLE
[Cancellation Form] Cancellation Isn't Confirmed 

### ISSUE THREAD

**Issue by souredoutlook** [getsentry/sentry#81971]
### Environment

SaaS (https://sentry.io/)

### Steps to Reproduce

1. Navigate to the subscription management page
2. Select "Cancel Subscription:"
3. Complete the flow
4. BUG no 1 observe that the subscription overview page does not show that there are scheduled changes to the subscription
5. Go back to the subscription management flow
6. Select "Cancel Subscription"
7. BUG no 2 Note that it's not possible to finish the flow 

### Expected Result

Cancelling the subscription the first time should update the UI to indicate that there are scheduled changes.

Cancelling the subscription should not show a "cancel subscription" button on the subscription management flow (because it's already cancelled)

### Actual Result

Without hard refreshing the app it's not possible to see that changes to the subscription have been scheduled once the cancellation flow is complete

Once the scheduled changes are available in the UI it's still possible to attempt to cancel your subscription again but the process  actually broken and can't be completed (giving the impression that we're preventing users from cancelling)

### Product Area

Settings - Subscription

### Link

https://sentry.zendesk.com/agent/tickets/140353

**Comment by shellyvo** [comment]
Another ticket here where they don't understand how to cancel and thought they did months ago: https://sentry.zendesk.com/agent/tickets/146948
